modules/dcache/src/test/java/diskCacheV111/util/PgpassTest.java

```diff
@@ -0,0 +1,55 @@
+package diskCacheV111.util;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+import static org.junit.Assert.assertNull;
+
+/**
+ * Some tests for the {@link Pgpass} class
+ *
+ * @author Tom Dittrich
+ * @version 0.1
+ */
+public class PgpassTest {
+
+    final String FILEPATH = "src/test/resources/pgpassTestFile.txt";
```

Comment:
The `src` directory in a maven project is reserved exclusively for checked out code.  All dynamically created files (of any type) belong in the `target` directory.

Perhaps a better approach for unit-testing is to use an in-memory filesystem and configure it (in the test) exactly as required.  This avoids any accidental problems (wrong permissions, file already exists, ...)

For examples on how to do this with JimFS, see:
modules/gplazma2-grid/src/test/java/org/dcache/gplazma/util/IGTFInfoFileTest.java
modules/gplazma2-multimap/src/test/java/org/dcache/gplazma/plugins/GplazmaMultiMapFileTest.java

Reply:
Sounds better, you are right.
Maybe it's a good idea for a future edu-issue? It can be combined with writing more tests or make the class better testable.

Reply:
I think this should be fixed before code get merged. something  like `File#createTempFile` will do the job.